Sources/TuistCore/NodeLoaders/BundleLoader.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+enum BundleLoaderError: FatalError, Equatable {
+    case bundleNotFound(AbsolutePath)
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .bundleNotFound:
+            return .abort
+        }
+    }
+
+    /// Error description
+    var description: String {
+        switch self {
+        case let .bundleNotFound(path):
+            return "Couldn't find bundle at \(path.pathString)"
+        }
+    }
+}
+
+public protocol BundleLoading {
+    /// Reads an existing bundle and returns its in-memory representation, `ValueGraphDependency.bundle`.
+    /// - Parameter path: Path to the .bundle.
+    func load(path: AbsolutePath) throws -> GraphDependency
+}
+
+public final class BundleLoader: BundleLoading {
```

Comment:
does this component need to be `TuistCore`? AFAICT, it can be in `TuistCache` only

Reply:
Maybe at some point we might add standalone bundle's as dependencies to targets. Like google maps framework makes you manuelly add its resources bundle to the project?

Reply:
that's fair but for the time being, I'd leave it in a framework where it's exclusively used at the moment. We can move it in the future if we deem it necessary 🙂